flocker/node/agents/blockdevice.py

```diff
@@ -551,6 +577,83 @@ def _valid_size(size):
 
 @implementer(IStateChange)
 @provider(IDatasetStateChangeFactory)
+class CreateMountSymlink(PClass):
+    """
+    Create a Symlink to the mount at the specified shared path location that we
+    will share with external agents using Flocker.
+
+    :ivar UUID dataset_id: The unique identifier of the dataset.
+    :ivar FilePath mountpoint: The filesystem location at which the backing
+        volume is mounted.
+    :ivar FilePath link_path: The path that will be created linking to the
+        mountpoint.
+    """
+    dataset_id = field(type=UUID, mandatory=True)
+    mountpoint = field(type=FilePath, mandatory=True)
+    link_path = field(type=FilePath, mandatory=True)
+
+    @classmethod
+    def from_state_and_config(cls, discovered_dataset, desired_dataset):
+        return cls(
+            dataset_id=desired_dataset.dataset_id,
+            mountpoint=discovered_dataset.mount_point,
+            link_path=desired_dataset.link_path,
+        )
+
+    @property
+    def eliot_action(self):
+        return CREATE_MOUNT_SYMLINK(_logger, dataset_id=self.dataset_id)
+
+    def run(self, deployer):
+        """
+        Create a symlink from the mounted blockdevice to a path that will given
+        to flocker clients. This layer of indirection is added so that we can
+        atomically create a mounted directory path.
+        """
+        parent = self.link_path.parent()
+        if not parent.exists():
```

Comment:
That seems problematic. There's a bunch of  permissios stuff you probably want to get right, for exampe. It rather seems like it should  just assume it's bee pre-created.